src/test/auxiliary/derive-no-std.rs

```diff
@@ -8,32 +8,33 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-#![feature(rand, collections, rustc_private)]
-#![no_std]
+// no-prefer-dynamic
```

Comment:
Yeah building a dylib requires lang items to be resolved, but as an rlib (what no-prefer-dynamic asks for) means that they can stay dangling.